#건강검진 대사 증후군 분석

In [10]:
import pandas as pd

In [11]:
## 우측하단 언어 python 주의 // 파일 리드 시 인코딩을 cp949로 할 것
df = pd.read_csv("../../Part-I/국민건강보험공단_건강검진정보_20211229.CSV", encoding='cp949')
df = df.dropna(axis=0) ## 결측값이 들어있는 행 전체 삭제하기 dropna(axis=0) // 결측값이 들어있는 열 전체 삭제하기 dropna(axis=1)
df


C:\Users\user\AppData\Local\Temp\ipykernel_10208\972869629.py:2: DtypeWarning: Columns (30) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("../../Part-I/국민건강보험공단_건강검진정보_20211229.CSV", encoding='cp949')


기준년도  가입자 일련번호  시도코드  성별코드  연령대 코드(5세단위)  신장(5Cm단위)  체중(5Kg 단위)  허리둘레  \
6    2020.0       7.0  44.0   1.0           9.0      165.0        80.0  93.0   
16   2020.0      17.0  30.0   1.0          11.0      160.0        65.0  90.0   
21   2020.0      22.0  41.0   2.0          11.0      155.0        50.0  65.9   
27   2020.0      28.0  11.0   1.0          17.0      155.0        60.0  82.0   
47   2020.0      48.0  26.0   1.0          16.0      165.0        65.0  94.0   
59   2020.0      60.0  41.0   2.0          12.0      150.0        55.0  89.0   
83   2020.0      84.0  41.0   1.0          10.0      170.0        70.0  85.9   
84   2020.0      85.0  26.0   1.0          11.0      170.0        75.0  88.0   
95   2020.0      96.0  47.0   2.0           9.0      160.0        50.0  69.0   
101  2020.0     102.0  28.0   2.0          10.0      175.0        75.0  76.5   
107  2020.0     108.0  47.0   2.0           9.0      165.0        55.0  70.0   
138  2020.0     139.0  31.0   1.0          12.0      170.0        70.0  88.1   
148  2020.0     149.0  11.0   2.0          15.0      150.0        50.0  80.5   
149  2020.0     150.0  41.0   1.0           9.0      170.0        65.0  84.2   
155  2020.0     156.0  41.0   1.0           9.0      165.0        70.0  85.0   
159  2020.0     160.0  11.0   2.0           9.0      165.0        55.0  66.0   
161  2020.0     162.0  11.0   1.0          13.0      165.0        65.0  89.0   
171  2020.0     172.0  41.0   1.0          13.0      165.0        65.0  82.6   
205  2020.0     206.0  28.0   1.0          12.0      160.0        60.0  75.3   
206  2020.0     207.0  43.0   2.0          15.0      150.0        65.0  96.0   
214  2020.0     215.0  44.0   1.0          11.0      165.0        60.0  75.0   
222  2020.0     223.0  28.0   1.0          12.0      165.0        60.0  75.4   
236  2020.0     237.0  41.0   2.0          13.0      150.0        45.0  70.2   
263  2020.0     264.0  30.0   2.0          14.0      150.0        50.0  77.0   
266  2020.0     267.0  41.0   1.0          10.0      165.0        55.0  77.8   
272  2020.0     273.0  28.0   2.0          15.0      145.0        50.0  73.8   
286  2020.0     287.0  11.0   2.0          13.0      155.0        45.0  68.0   
292  2020.0     293.0  45.0   1.0          10.0      160.0        60.0  83.2   

     시력(좌)  시력(우)  ...  혈청크레아티닌  (혈청지오티)AST  (혈청지오티)ALT  감마 지티피  흡연상태  음주여부  \
6      0.8    0.7  ...      0.7        18.0        20.0    35.0   3.0   1.0   
16     0.6    0.6  ...      1.0        29.0        54.0    24.0   1.0   1.0   
21     0.7    1.0  ...      0.8        24.0        14.0    12.0   1.0   0.0   
27     0.5    0.7  ...      1.4        33.0        22.0    29.0   1.0   1.0   
47     1.5    0.6  ...      1.0        28.0        20.0    43.0   1.0   1.0   
59     1.0    0.8  ...      0.9        24.0        37.0    40.0   1.0   1.0   
83     1.0    1.0  ...      0.9        21.0        22.0    35.0   1.0   1.0   
84     1.0    0.6  ...      1.1        20.0        22.0    23.0   3.0   1.0   
95     1.2    1.2  ...      0.8        25.0        19.0    16.0   1.0   1.0   
101    0.7    0.4  ...      0.7        20.0        15.0    29.0   1.0   0.0   
107    0.9    1.0  ...      0.6        18.0        12.0    12.0   1.0   1.0   
138    0.8    0.7  ...      0.9        30.0        34.0    34.0   2.0   1.0   
148    0.7    0.3  ...      0.9        21.0        16.0    17.0   1.0   0.0   
149    1.0    1.2  ...      1.0        21.0        27.0    19.0   2.0   1.0   
155    1.0    1.2  ...      1.1        20.0        33.0    18.0   1.0   1.0   
159    0.9    0.8  ...      0.7        18.0         9.0    13.0   1.0   1.0   
161    1.0    1.0  ...      1.4        23.0        19.0    54.0   1.0   1.0   
171    1.0    0.9  ...      1.1        23.0        28.0    31.0   3.0   1.0   
205    0.6    0.7  ...      1.0        28.0        33.0    22.0   1.0   0.0   
206    1.0    1.0  ...      0.8        40.0        45.0    57.0   1.0   0.0   
214    1.0    1.0  ...     

 + 열명확인 및 변경

In [12]:
df.columns

Index(['기준년도', '가입자 일련번호', '시도코드', '성별코드', '연령대 코드(5세단위)', '신장(5Cm단위)',
       '체중(5Kg 단위)', '허리둘레', '시력(좌)', '시력(우)', '청력(좌)', '청력(우)', '수축기 혈압',
       '이완기 혈압', '식전혈당(공복혈당)', '총 콜레스테롤', '트리글리세라이드', 'HDL 콜레스테롤', 'LDL 콜레스테롤',
       '혈색소', '요단백', '혈청크레아티닌', '(혈청지오티)AST', '(혈청지오티)ALT', '감마 지티피', '흡연상태',
       '음주여부', '구강검진 수검여부', '치아우식증유무', '치석', '데이터 공개일자'],
      dtype='object')

In [17]:
# df를 리스트로 형변환
cc = list(df.columns)
cc

['기준년도',
 '가입자 일련번호',
 '시도코드',
 '성별코드',
 '연령대 코드(5세단위)',
 '신장(5Cm단위)',
 '체중(5Kg 단위)',
 '허리둘레',
 '시력(좌)',
 '시력(우)',
 '청력(좌)',
 '청력(우)',
 '수축기 혈압',
 '이완기 혈압',
 '식전혈당(공복혈당)',
 '총 콜레스테롤',
 '트리글리세라이드',
 'HDL 콜레스테롤',
 'LDL 콜레스테롤',
 '혈색소',
 '요단백',
 '혈청크레아티닌',
 '(혈청지오티)AST',
 '(혈청지오티)ALT',
 '감마 지티피',
 '흡연상태',
 '음주여부',
 '구강검진 수검여부',
 '치아우식증유무',
 '치석',
 '데이터 공개일자']

In [18]:
for c in cc :
  print(c)

기준년도
가입자 일련번호
시도코드
성별코드
연령대 코드(5세단위)
신장(5Cm단위)
체중(5Kg 단위)
허리둘레
시력(좌)
시력(우)
청력(좌)
청력(우)
수축기 혈압
이완기 혈압
식전혈당(공복혈당)
총 콜레스테롤
트리글리세라이드
HDL 콜레스테롤
LDL 콜레스테롤
혈색소
요단백
혈청크레아티닌
(혈청지오티)AST
(혈청지오티)ALT
감마 지티피
흡연상태
음주여부
구강검진 수검여부
치아우식증유무
치석
데이터 공개일자


In [ ]:
# 위치번호와 정보가 같이 받아짐
for c in enumerate(cc) :
  print(c)



In [21]:
# 위치번호와 정보가 따로 받아짐
for i, c in enumerate (cc) :
  print(i, c)

0 기준년도
1 가입자 일련번호
2 시도코드
3 성별코드
4 연령대 코드(5세단위)
5 신장(5Cm단위)
6 체중(5Kg 단위)
7 허리둘레
8 시력(좌)
9 시력(우)
10 청력(좌)
11 청력(우)
12 수축기 혈압
13 이완기 혈압
14 식전혈당(공복혈당)
15 총 콜레스테롤
16 트리글리세라이드
17 HDL 콜레스테롤
18 LDL 콜레스테롤
19 혈색소
20 요단백
21 혈청크레아티닌
22 (혈청지오티)AST
23 (혈청지오티)ALT
24 감마 지티피
25 흡연상태
26 음주여부
27 구강검진 수검여부
28 치아우식증유무
29 치석
30 데이터 공개일자


In [15]:
df.index

Int64Index([  6,  16,  21,  27,  47,  59,  83,  84,  95, 101, 107, 138, 148,
            149, 155, 159, 161, 171, 205, 206, 214, 222, 236, 263, 266, 272,
            286, 292],
           dtype='int64')

 + 필요한 열 가져오기

In [38]:
df2 = df[['성별코드', '허리둘레', '수축기 혈압','이완기 혈압','식전혈당(공복혈당)' ,'트리글리세라이드', 'HDL 콜레스테롤']]
df2.head()


성별코드  허리둘레  수축기 혈압  이완기 혈압  식전혈당(공복혈당)  트리글리세라이드  HDL 콜레스테롤
6    1.0  93.0   112.0    73.0       250.0     265.0       26.0
16   1.0  90.0   138.0    88.0        94.0     328.0       46.0
21   2.0  65.9   114.0    73.0        94.0      85.0       55.0
27   1.0  82.0   139.0    85.0       104.0      75.0       57.0
47   1.0  94.0   110.0    72.0       114.0     106.0       53.0

In [39]:
lt = list(df2.columns)
lt

['성별코드', '허리둘레', '수축기 혈압', '이완기 혈압', '식전혈당(공복혈당)', '트리글리세라이드', 'HDL 콜레스테롤']

In [40]:
# 리스트 내 공백제거 *****중요*****
lt = [c.replace(' ', '') for c in lt]
lt

['성별코드', '허리둘레', '수축기혈압', '이완기혈압', '식전혈당(공복혈당)', '트리글리세라이드', 'HDL콜레스테롤']

In [42]:
lt[4] = '공복혈당'
df2.columns = lt
df2.head()

성별코드  허리둘레  수축기혈압  이완기혈압   공복혈당  트리글리세라이드  HDL콜레스테롤
6    1.0  93.0  112.0   73.0  250.0     265.0      26.0
16   1.0  90.0  138.0   88.0   94.0     328.0      46.0
21   2.0  65.9  114.0   73.0   94.0      85.0      55.0
27   1.0  82.0  139.0   85.0  104.0      75.0      57.0
47   1.0  94.0  110.0   72.0  114.0     106.0      53.0

In [43]:
import numpy as np

 + 높은 혈압 (130/85mmHg 이상)
 + dfb$높은혈압 <- ((dfb$수축기혈압 >= 130) | (dfb$이완기혈압 >= 85))

 + 높은 혈당 (공복 혈당 100mg/dL 이상)
 + dfb$높은혈당 <- ((dfb$공복혈당 >= 100))

 + 높은 중성지방(트리글리세라이드 150mg/dL 이상)
 + dfb$높은중성지방 <- (dfb$트리글리세라이드 >= 150)

 + 낮은 HDL 콜레스테롤 수치(남성은 40mg/dL 미만, 여성은 50mg/dL 미만)
 + dfb$낮은콜레스테롤 <- ((dfb$성별코드 == 1 & dfb$HDL콜레스테롤 < 40) | 
                  (dfb$성별코드 == 2 & dfb$HDL콜레스테롤 < 50))

 + 복부 비만(남성 90cm 이상, 여성 85cm 이상)
 + dfb$복부비만 <- ((dfb$성별코드 == 1 & dfb$허리둘레 >= 90) | 
               (dfb$성별코드 == 2 & dfb$허리둘레 >= 85))

In [56]:
df2['높은혈압'] = np.where((df2['수축기혈압'] >= 130) | (df2['이완기혈압'] >= 85), 1, 0)
df2['높은혈당'] = np.where((df2['공복혈당'] >= 100), 1, 0)
df2['높은중성지방'] = np.where(((df2['성별코드'] == 1 ) & (df2['HDL콜레스테롤'] < 40))  | ((df2['성별코드'] == 2 ) & (df2['HDL콜레스테롤'] < 50)))


C:\Users\user\AppData\Local\Temp\ipykernel_10208\1043334712.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['높은혈압'] = np.where((df2['수축기혈압'] >= 130) | (df2['이완기혈압'] >= 85), 1, 0)
C:\Users\user\AppData\Local\Temp\ipykernel_10208\1043334712.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['높은혈당'] = np.where((df2['공복혈당'] >= 100), 1, 0)


ValueError: Length of values (1) does not match length of index (28)